controllers/automation/policyautomation_controller.go

```diff
@@ -195,6 +196,29 @@ func (r *PolicyAutomationReconciler) getViolationContext(
 		return violationContext, nil
 	}
 
+	// targetList is noncompliant list, this step compares compliant
+	// so that check any changes happened during reconcile
+	for _, clusterName := range targetList {
```

Comment:
I don't think this block of code is required. If there is a mismatch in the status in the root policy and replicated policy, it means the policy on the cluster has become compliant and the root policy status will soon be updated or already has but the cached value in memory is stale.

Regardless, the cluster became noncompliant so that should still trigger an AnsibleJob. For example, if a policy goes non-compliant and then quickly to compliant and your Ansible job creates a ServiceNow ticket to notify a systems administrator, you'd still want the ServiceNow ticket created.

Reply:
This is not critical but I assumed maybe this is unclear(confusing) for the customers, cuz policyautomation says noncompliant but ansiblejob is compliant. also, I would like to avoid creating useless ansiblejob as much as possible. can delete it if u want